File: IUT/2A/CPOA/TP_noté/automate/Automate.cpp
#include "Automate.h"

// Constructors/Destructors
//  

Automate::Automate () {
initAttributes();
}

Automate::~Automate () { }

//  
// Methods
//  


// Accessor methods
//  


/**
 * Set the value of m_etatcourant
 * @param new_var the new value of m_etatcourant
 */
void Automate::setEtatCourant (IEtat * new_var) {
  m_etatcourant = new_var;
}

/**
 * Get the value of m_etatcourant
 * @return the value of m_etatcourant
 */
IEtat * Automate::getEtatCourant () {
  return m_etatcourant;
}

// Other methods
//  

void Automate::initAttributes () {
  m_etatcourant = new IEtat();
}

File: IUT/2A/CPOA/TP6/umbrello/Automate.java

import java.util.*;

public class Automate implements IEvenement {

  //
  // Fields
  //

  private Chronometre controle;

  public IEtat m_etatCourant;  public Automate () { };
  
  //
  // Methods
  //


  //
  // Accessor methods
  //

  /**
   * Set the value of controle
   * @param newVar the new value of controle
   */
  private void setControle (Chronometre newVar) {
    controle = newVar;
  }

  /**
   * Get the value of controle
   * @return the value of controle
   */
  private Chronometre getControle () {
    return controle;
  }

  /**
   * Set the value of m_etatCourant
   * @param newVar the new value of m_etatCourant
   */
  public void setEtatCourant (IEtat newVar) {
    m_etatCourant = newVar;
  }

  /**
   * Get the value of m_etatCourant
   * @return the value of m_etatCourant
   */
  public IEtat getEtatCourant () {
    return m_etatCourant;
  }

  //
  // Other methods
  //

  /**
   * @param        etat
   */
  public void changementEtat(IEtat etat)
  {
  }


  /**
   */
  public void evenement1()
  {
  }


  /**
   */
  public void evenement2()
  {
  }


}
